reconcile/glitchtip/integration.py

```diff
@@ -106,6 +118,17 @@ def fetch_desired_state(
                         )
                     )
 
+            # Get users via ldap
+            for ldap_group in glitchtip_team.ldap_groups or []:
+                for member in internal_groups_client.group(ldap_group).members:
+                    users.append(
+                        User(
+                            email=f"{member.id}@{mail_domain}",
```

Comment:
there is the ` get_group_members(self, groups_dns: set[str])` function in ldap_client.py you can adapt for your needs for sure

Reply:
> Will there be a case that member.id has duplicate values? Either with other group or role_user.org_username above?

Yes, that's possible. Fixed in [148461e](https://github.com/app-sre/qontract-reconcile/pull/3842/commits/148461e29d73eff31df48f6a757d9094c618a1e2)

> And is there a way to batch fetch and cache ldap group member info, here it will send many requests out in a loop.

Unfortunately, the internal groups API doesn't support fetching multiple groups in one API call :( We may need to cache the results in the future, but this shouldn't be a problem for now.

Reply:
> there is the  get_group_members(self, groups_dns: set[str]) function in ldap_client.py you can adapt for your needs for sure

Ah, I wasn't aware about `LdapClient`. I'll take a look

Reply:
For the deduplicate logic, what if they have same email but different role?

Reply:
In this case, the first one (from roles) wins. This also opens the option to override an LDAP user and specify another glitchtip role.

Reply:
Ad `LdapClient`, I keep it as it is with `InternalGroups`. `LdapClient.get_group_members` needs the group full qualified in LDAP notation and I think just using the group name is much easier for our tenants. `cn=group-name,ou=foo,ou=bar,ou=whatever` vs `group-name`

Reply:
> In this case, the first one (from roles) wins. This also opens the option to override an LDAP user and specify another glitchtip role.

You mean the last one wins, right? According to current code, ldap_groups will override roles ones.

Reply:
> much easier for our tenants
i agree that tenants should not necessarily need to know about this. the suffix for the full DN would need to be added behind the scenes. for now, looking up those groups in a loop is not a problem in my opinion. lets reconsider once we reach adoption that scale becomes important. leave a comment for future us :)

Reply:
> You mean the last one wins, right? According to current code, ldap_groups will override roles ones.

yes, 🤦. I changed it in the last commit and made it more explicit.

Reply:
https://github.com/app-sre/qontract-reconcile/pull/3842/files#diff-23b401188c8043f58c41e534c681e5dc607cb9eaebd981295ce1e4e49b97a21dR129

`user not in users` check is inside another 2 nested loops, the whole append will be O(N^2).

Can use your previous structure to make it faster `{email: user}` or just `set[User]` since `__hash__` is already implemented for `email` field.

Reply:
Changed it to `set(users)`

Reply:
`set(users)` will implicit ignore later ones in sequence, at least that's CPython behaviour, maybe worth adding a comment for which order wins. Other than that, all good!